### DISCUSSION TITLE
Option to allow mouse drag & drop to be move instead of copy

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1693] [Suggestions]
`INTRO`
I just realized that although this topic has been brought up quite a bit, there is still no official “Discussions - Suggestions” based post about it.  Only a topic under “Issues” but this is a feature request not an issue which I have mentioned in the past.  So I decided to take some time  to make this official post here.
If possible PLEASE do not close this, as I have spent a good amount of time to create this.
I know I have made many feature request, but I think I would say that this one is still one of the most desired to have added to DC.  

`REQUEST`
Add OPTION to allow mouse drag & drop to be able to be set to MOVE instead of COPY.

`OPTIONS UPDATES`
Options - Mouse - Drag & Drop
This section actually already exists in DC Options with only one lonely little setting in it.
This request could be added to this section.

Default mouse drag & drop action: MOVE / COPY

Can add a little warning message if feel that it is necessary.
Note: By setting this action to MOVE, all drag & drop actions including to other drives will be MOVE instead of COPY.

`OTHER ITEMS THAT MAY NEED TO BE UPDATED IF THIS IS ADDED`
D&D actions this would apply to.....
-File pane to File Pane
-File pane to Tabs
-File pane to Address Bar (if my suggestion for this is added in the future)

Modifier Keys....
Because DC uses COPY action by default, the decision has been made to allow
Ctrl+LMB D&D
Shift+LMB D&D 
to both perform the MOVE action
if this feature is added to allow MOVE to be the default action, then these two modifier actions would just need to be set to both be COPY.

Thank You to anyone who reads this and to anyone willing to give this a try.

**Comment by advice2020** [comment]
I have more to say about this because I imagine there will be some comments, so I decided to put here rather than in the main request post itself.  I know this is a long post, but mainly because it appears that there are some users that seem to be so against this, and I do not understand why, as it would just be an OPTION.

`THINGS TO KEEP IN MIND`
-None of this is meant to offend anyone, just trying to understand / sharing some opinions.
-This would be added as an OPTION for user to be able to CHOOSE if they want this or not.  All current default settings would remain EXACTLY the same, so users who prefer the current default will not even notice this at all.

`THINGS I AM AWARE OF`
-I am fully aware that DC contains modifiers Ctrl & Shift that allow this MOVE action
-I am aware that if you use RMB D&D that there is a context menu
-I know that DC is considered an “orthodox” file manager

`QUESTIONS / OPINIONS`
-As mentioned above, just because DC is more of an orthodox file manager, it still uses mouse actions, and it also probably has features that are not specifically OFM specific, so I am not sure why this is brought up so defensively when users request none OFM features.
Especially this one that will in no way affect any users that do not want to use it.
Does DC have to only contain OFM based features?
-My view is that when using a computer, wouldn't you want to have the least amount of requirements to perform an action?  Not sure why OFM would be used as a reason to not add something like this (again OPTION).
-I think OFMs have great features/workflows I wouldn't be using DC if I didn't, but there are certain scenarios where D&D actions are more efficient then keyboard approach, where having to keep going into destination file pane to set where F6 will move files into.  Situations where you are moving on 1 or 2 files at a time.  (I can give some full examples but this post will be even longer and it is not the point of this request)
-If users do not want to use mouse action, that is fine they do not have to, but nice to know that they are available if needed.
-I would think that users are more likely to MOVE files then they would COPY them in a D&D action
-Not just FMs, but most programs that require mouse, support D&D to move things around, so it is a common computer based action.
-When performing this D&D action over & over using modifiers, if you are trying to do this fast, you have to have some good timing & coordination, requires a lot of focus, as opposed to just a simple D&D.
-Sometimes I have to use my DEs default FM if organizing many files that require D&D, but then I end up missing out on all the amazing features that DC offers.

`LINKS`
Here are links to other posts that mention this feature request
DC Forum
https://doublecmd.h1n.ru/viewtopic.php?t=3354
https://doublecmd.h1n.ru/viewtopic.php?t=7611
Github (Under “Issues”)
#271

Hopefully some users will agree with some of the information I have shared above and hopefully this was worth the effort.
Again thank you to anyone who actually reads all of this

**Comment by alexx2000** [comment]
Done, option `DefaultDropEffect` in the `doublecmd.xml`, `True - Copy`, `False - Move`.

**Comment by ags1234** [comment]
Just out of curiosity, are there more options in config file that do not have a correspondent in the GUI?

**Reply by alexx2000** [reply]
https://doublecmd.github.io/doc/en/configxml.html#manualed

**Comment by advice2020** [comment]
Alexander!!!! this is amazing!!!!!
I stopped by this site randomly today and was so excited to see this, had to compile right away to try this.
This one was at the top of my list, thank you so much for adding this, this works great!

And if this wasn't enough, this also brought my attention to "Options to change manually".
I remember looking into this briefly in the past but no settings stood out to me, but after looking into this with more detail I found some great settings.
Sorry but there is especially one I just wanted to mention.  In the past I was going to request this but never did because I did not think anyone would really care about it.
`<RenameConfirmMouse>`
For any Linux/Windows users unaware of this one, if you set this to `True` it mimics Windows file explorer behavior of being able to click outside the name edit field to apply the changes when using F2 inline renaming.
I wish I had asked about this in the past, I cannot tell you how many times I lost renames by clicking outside the edit field by habit, and the longer the filenames I typed, the more often I did it :)

Anyways, thank you again for this Alexander I really appreciate this.
I really still cannot believe how great this program.

**Reply by VanCamp-K** [reply]
Seconding this sentiment; manually edited config options addressed the few remaining qualms I had while using doublecmd. Tyvm, @alexx2000 !🙏

For any other GNU-Linux users, the current version of Doublecmd available on the mintinstall Software Manager (1.1.11-1build1, as of Jan 30 '26) does NOT include at least some of the manual config options mentioned above. Downloading & unpacking a more recent [release](https://github.com/doublecmd/doublecmd/releases) is all you need to access the updated options.
